### PR TITLE
refactor(primitives): don't duplicate the SpecId::enabled() logic

### PR DESCRIPTION
**Pull request by IaroslavMazur**
* remove unnecessary function/trait implementations

* fix docs errors

### CODE REVIEW

File: crates/primitives/src/db.rs [hunk 1]
```diff
@@ -61,13 +61,6 @@ pub trait DatabaseRef {
 #[derive(Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct WrapDatabaseRef<T: DatabaseRef>(pub T);
 
-impl<F: DatabaseRef> From<F> for WrapDatabaseRef<F> {
-    #[inline]
-    fn from(f: F) -> Self {
-        WrapDatabaseRef(f)
-    }
-}
-
```
Comment (rakita) [inline crates/primitives/src/db.rs:70]: Revert this, in general, it is useful outside of lib

File: crates/primitives/src/specification.rs [hunk 2]
```diff
@@ -35,11 +35,6 @@ pub enum SpecId {
 }
 
 impl SpecId {
-    #[inline]
-    pub fn try_from_u8(spec_id: u8) -> Option<Self> {
-        Self::n(spec_id)
-    }
```
Comment (rakita) [inline crates/primitives/src/specification.rs:41]: Revert this, transitioning from u8 can be useful outside of lib

File: crates/primitives/src/specification.rs [hunk 3]
```diff
@@ -111,7 +106,7 @@ pub trait Spec: Sized {
             }
         }
 
-        Self::SPEC_ID as u8 >= spec_id as u8
+        SpecId::enabled(Self::SPEC_ID, spec_id)
```
Comment (rakita) [inline crates/primitives/src/specification.rs:109]: This is great, and you can even remove previous lines as they are checked inside `SpecId::enabled`

**Review by rakita — CHANGES_REQUESTED** [review]
Mostly revert public function they should stay, and remove few more lines in `Spec::enabled`

File: crates/primitives/src/specification.rs (hunk 3 above)
Comment (IaroslavMazur) [reply crates/primitives/src/specification.rs:99]: Great finding!

**Review by rakita — APPROVED** [review]
lgtm

**Pull request merged**